### PR TITLE
[opt](cache) Support segment cache dynamic opening and closing

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Dynamically modify the config to clear the cache, each time the disable cache will only be cleared once.
TODO, Support page cache and other caches.

`curl -X POST http://xxxx:8040/api/update_config?disable_segment_cache=true`

image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.19 seconds
 stream load tsv:          537 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162041419 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.31 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162253640 Bytes

File: be/src/runtime/memory/cache_manager.cpp [hunk 1]
```diff
@@ -43,7 +43,16 @@ int64_t CacheManager::for_each_cache_prune_stale(RuntimeProfile* profile) {
 
 int64_t CacheManager::for_each_cache_prune_all(RuntimeProfile* profile) {
     return for_each_cache_prune_stale_wrap(
-            [](CachePolicy* cache_policy) { cache_policy->prune_all(); }, profile);
+            [](CachePolicy* cache_policy) { cache_policy->prune_all(false); }, profile);
+}
+
+void CacheManager::clear_once(const std::string& name) {
```
Comment (yiguolei) [inline be/src/runtime/memory/cache_manager.cpp:49]: using  enum  instead of name.
name is very tricky and may be spell error or typo errors.

Comment (xinyiZzz) [reply be/src/runtime/memory/cache_manager.cpp]: fix

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.26 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162292976 Bytes

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.7 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162096344 Bytes

File: be/src/olap/page_cache.h [hunk 2]
```diff
@@ -103,21 +103,22 @@ class StoragePageCache {
     class DataPageCache : public LRUCachePolicy {
     public:
         DataPageCache(size_t capacity, uint32_t num_shards)
-                : LRUCachePolicy("DataPageCache", capacity, LRUCacheType::SIZE,
+                : LRUCachePolicy(CachePolicy::Type::DATA_PAGE_CACHE, capacity, LRUCacheType::SIZE,
```
Comment (yiguolei) [inline be/src/olap/page_cache.h:106]: rename to CACHE_TYPE
CachePolicy::TYPE  user may treat it as policy type. But actually, it is one kind of cache, not one kink of policy like lru,fifio...

Comment (xinyiZzz) [reply be/src/olap/page_cache.h]: done

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 49.12 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162509073 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**